generator/operation.go

```diff
@@ -739,8 +743,12 @@ func (b *codeGenOpBuilder) MakeHeaderItem(receiver, paramName, indexVar, path, v
 	res.Name = paramName
 	res.Path = path
 	res.Location = "header"
-	res.ValueExpression = valueExpression
+	res.ValueExpression = golang.MangleVarName(swag.ToGoName(valueExpression))
 	res.CollectionFormat = items.CollectionFormat
+	if res.CollectionFormat == "" && items.Items != nil {
+		// When not defined,CollectionFormat defaults to csv
+		res.CollectionFormat = "csv"
```

Comment:
not a fan of this because it now sprinkles this across the codebase and people have to remember to do this. Better to postpone this until the last responsible moment

Reply:
ok.